src/plone/api/content.py

```diff
@@ -473,3 +474,45 @@ def get_uuid(obj=None):
     :Example: :ref:`content_get_uuid_example`
     """
     return IUUID(obj)
+
+
+def find(context=None, depth=None, **kwargs):
+    """Find content
+
+    :param context: Context within we want to search
+    :type obj: Content object
+    :param depth: How far in the content tree we want to search from context
+    :type obj: Content object
+    :returns: Catalog brains
+    :rtype: List
+    :Example: :ref:`content_find_example`
+    """
+    query = {}
+    query.update(**kwargs)
+
+    # Passing a context or depth overides the exising path query
+    if context or depth:
+        query['path'] = {}
+
+    # Limit search depth
+    if depth is not None:
+        # If we don't have a context, we'll assume the portal root.
+        if context is None:
```

Comment:
It's just a mapping to catalog.searchResults, which has the same behaviour.

If you pass depth you'll need a context, because depth is part of the path query.

object_provides is just another index, but normally you'll have to pass the \__identifier__. I don't like this, so that's what the code for object_provides is doing.

No argument is required, because they just map to indexes, which may or may not be there :)